### ISSUE TITLE
Can presto add catalog without restart using rest api?

### ISSUE THREAD

**Issue by ltylty** [trinodb/trino#2110]
I'm doing olap system using java.One job is add external datasource to analysis.I want to use presto by sql to query many external datasources.The best way for me is using rest api to add connector.

**Comment by findepi** [comment]
@ltylty currently it's not possible.

**Comment by NurtureLabs** [comment]
I'd like to put in the time to work on this. Can someone guide me on this?

**Comment by scirag** [comment]
> I'd like to put in the time to work on this. Can someone guide me on this?

+1

**Comment by allendang001** [comment]
> @ltylty currently it's not possible.

Could you tell me what is the reason?

**Comment by huzekang** [comment]
I also need this function

**Comment by tooptoop4** [comment]
https://openlookeng.io/docs/docs/admin/dynamic-catalog.html / https://github.com/openlookeng/hetu-core might have some inspiration

**Comment by cxl-todd** [comment]
Any movement on this? 

I think reloading is only a portion of the requirements I have. I need to be able to manage the catalog of datasources externally. Users can add new datasources at runtime. I'd ideally like trino to be able to utilize an external catalog that is dynamically updated.

**Comment by ribrewguy** [comment]
I created a naive approach at hot reloading the property files. A PR is here https://github.com/trinodb/trino/pull/12249

I got blocked by trying to test and build it. Testing in the IDE won't work as it seems some classes are supposed to be generated in the parser module. Running it from the command line to (hopefully) build those files fails with ``` Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:3.3.0:analyze-only (default) on project trino-main: Dependency problems found ```

Any opinion on the PR or how to get past the mvn issue would be greatly appreciated.
